### PR TITLE
avoid wasting space in parseScriptTemplate() and fix some typo

### PR DESCRIPTION
**Pull request by qshuai**
PTAL

### CODE REVIEW

File: txscript/script.go [hunk 1]
```diff
@@ -275,7 +275,9 @@ func parseScriptTemplate(script []byte, opcodes *[256]opcode) ([]parsedOpcode, e
 		retScript = append(retScript, pop)
 	}
 
-	return retScript, nil
+	ret := make([]parsedOpcode, len(retScript))
```
Comment (stevenroose) [inline txscript/script.go]: This doesn't fall under the description of the PR.. I also don't really think it's necessary.

Comment (qshuai) [reply txscript/script.go]: This is a real P2SH transaction sample:
txid: 5da708e9a903ac5d41a0f3b0731bedf3a2f99522d9978ec4d7d93412157296ce
P2SH script:a914f323f8af174e6829a9c8acf1013f65f18370380987

I will output origin script length and final length of the  result slice in parseScriptTemplate() function. as following:

```
func parseScriptTemplate(script []byte, opcodes *[256]opcode) ([]parsedOpcode, error) {
fmt.Println("script length:", len(script))
         ....
fmt.Println("retScript length:", len(retScript))
return retScript, nil
```

And I using a unit testing here:
```
func TestRealScript(t *testing.T) {
	b, _ := hex.DecodeString("a914f323f8af174e6829a9c8acf1013f65f18370380987")
	_, err := parseScript(b)
	if err != nil {
		t.Error(err)
	}
}
```

And run the unit testing, output:
   script length: 23
   retScript length: 3

So I think there are the memory space of twenty structs(parsedOpcode) wasting!

If there is any error, please let me know.

Comment (qshuai) [reply txscript/script.go]: @stevenroose PTAL

Comment (stevenroose) [reply txscript/script.go]: @qshuai I'm saying that the PR says "error annotation", while this addresses something else.

Looking at the entire method, it does seem to make sense to do the copy, but please don't mix several things in one PR.

Comment (qshuai) [reply txscript/script.go]: @stevenroose Yeah,  you are right. This PR include several commits. Github will merge all my commits in one PR. 
Is there a good practice? 

I am a newer bitcoin coder, and my english is poor. But I just want learn more. Please pardon my misnomer if has.🙂

**Comment by stevenroose** [comment]
@qshuai You can squash these 4 commits as follows:
```
git rebase -i HEAD~4
```
Then in all the lines except the top one (the earliest commit), replace `pick` with `squash` (or `s`) and save. Then an editor will open to let you write a commit message for your squashed commit.

Then you can overwrite your branch here like this:
```
git push --force
```

**Comment by qshuai** [comment]
@stevenroose Thanks for your directions!

**Comment by qshuai** [comment]
@Roasbeef I close this pr first. If necessary, will reopen!

**Pull request closed without merging**